README.md

```diff
@@ -31,14 +34,378 @@ class AppKernel extends Kernel
+]
+```
+- InvalidRequestBodyFormatException
+
+response body:
+```php
+[
+    'title' => 'Invalid request body format', 
+    'statusCode' => 400, 
+    'type' => 'invalid_request_body_format', 
+    'details' => [
+        'cause': 'unexpected_value'
+    ]    
+]
+```
+- InvalidRequestContentTypeException
+
+response body:
+```php
+[
+    'title' => 'Invalid request content type', 
+    'statusCode' => 400, 
+    'type' => 'invalid_request_content_type', 
+    'details' => [
+        'code': 'unsupported'
+    ]    
+]
+```
+- InvalidRequestException
+
+response body:
+```php
+[
+    'title' => 'Invalid request', 
+    'statusCode' => 400, 
+    'type' => 'invalid_request', 
+]
+```
+- ValidationErrorException
+
+response body:
+```php
+[
+    'title' => 'Validation error', 
+    'statusCode' => 400, 
+    'type' => 'validation_error', 
+    'details' => [
+        'violations' => 'Invalid customField'
+    ]
+]
+```
+
+### Enable support security exceptions
+- php
+```php
+<?php
+// app/config/packages/security.php
+
+//..
+'firewalls' => [
+    //..
+    'main' => [
+        //..
+        'entry_point' => 'VisualCraft\RestBaseBundle\Security\AuthenticationEntryPoint',
+        //..
+    ],
+    //..
+],
+//..
+```
+- yaml
+```yaml
+security:
+    firewalls:
+        main:
+            entry_point: 'VisualCraft\RestBaseBundle\Security\AuthenticationEntryPoint'
+            //..
+```
+### Support custom exception
+You can create and add your own exceptions and convertors for them.
+
+- create your exception
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace App\Exceptions;
+
+
+use Throwable;class CustomException extends \RuntimeException
+{
+    private string $customField;
+    
+    public function __construct(string $customField, $message = "",$code = 0,Throwable $previous = null)
+    {
+        parent::__construct($message,$code,$previous);
+        $this->customField = $customField;
+    }
+    
+    public function getCustomField(): string
+    {
+        return $this->customField;
+    }
+}
+```
+
+- create converter
+```php
+<?php
+//app/src/Problem/ExceptionToProblemConverters/InvalidRequestBodyFormatExceptionConverter.php
+
+declare(strict_types=1);
+
+namespace VisualCraft\RestBaseBundle\Problem\ExceptionToProblemConverters;
+
+use Symfony\Component\HttpFoundation\Response;
+use VisualCraft\RestBaseBundle\Problem\ExceptionToProblemConverterInterface;
+use VisualCraft\RestBaseBundle\Problem\Problem;
+
+class CustomExceptionConverter implements ExceptionToProblemConverterInterface
+{
+    public function convert(\Throwable $exception): ?Problem
+    {
+        if (!$exception instanceof CustomException) {
+            return null;
+        }
+
+        $result = new Problem(
+            'Custom exception title',
+            Response::HTTP_BAD_REQUEST,
+            'custom_exception_type'
+        );
+        $result->addDetails('cause', 'custom exception cause');
+
+        return $result;
+    }
+}
+```
+
```

Comment:
https://symfony.com/doc/current/templating/twig_extension.html#register-an-extension-as-a-service
if you do not use Autoconfiguration